### PR TITLE
Add agent capability for efs support

### PR DESCRIPTION
**Pull request by yumex93**
### Summary

1. Add agent capability for efs support. Since the efs will only be supported on linux, the capability logic is only added to linux go file.
2. Remove some redundant test cases, use one test case to just verify all capabilities for the latest agent version. 

### Testing

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/app/agent_capability_unix.go [hunk 1]
```diff
@@ -114,3 +114,7 @@ func (agent *ecsAgent) appendAppMeshCapabilities(capabilities []*ecs.Attribute)
 func (agent *ecsAgent) appendTaskEIACapabilities(capabilities []*ecs.Attribute) []*ecs.Attribute {
 	return appendNameOnlyAttribute(capabilities, attributePrefix+taskEIAAttributeSuffix)
 }
+
+func (agent *ecsAgent) appendEFSCapabilities(capabilities []*ecs.Attribute) []*ecs.Attribute {
```
Comment (petderek) [inline agent/app/agent_capability_unix.go]: can you add a TODO here to check for the correct bind mount?

**Comment by sharanyad** [comment]
is efs capability the only way backend could know if an agent supports it? 
or do we pass some info in RCI (like GPU) that could help identify efs support?

**Comment by yumex93** [comment]
Yeah. This will be the only place. What the reason behind GPU passing info through RCI?

**Comment by adnxn** [comment]
closing in favor of https://github.com/aws/amazon-ecs-agent/pull/2248

**Pull request closed without merging**